packages/next/src/build/webpack/loaders/lightningcss-loader/src/loader.ts

```diff
@@ -0,0 +1,176 @@
+  return {}
+}
+
+const LOADER_NAME = `lightningcss-loader`
+export async function LightningCssLoader(
+  this: LoaderContext<ILightningCssLoaderConfig>,
+  source: string,
+  prevMap?: Record<string, any>
+): Promise<void> {
+  const done = this.async()
+  const options = this.getOptions()
+  const { implementation, targets: userTargets, ...opts } = options
+
+  options.modules ??= {}
+
+  if (implementation && typeof implementation.transformCss !== 'function') {
+    done(
+      new TypeError(
+        `[${LOADER_NAME}]: options.implementation.transformCss must be an 'lightningcss' transform function. Received ${typeof implementation.transformCss}`
+      )
+    )
+    return
+  }
+
+  const exports: CssExport[] = []
+  const imports: CssImport[] = []
+  const api: ApiParam[] = []
+  const replacements: ApiReplacement[] = []
+
+  if (options.modules?.exportOnlyLocals !== true) {
+    imports.unshift({
+      type: 'api_import',
+      importName: '___CSS_LOADER_API_IMPORT___',
+      url: stringifyRequest(
+        this,
+        require.resolve('../../css-loader/src/runtime/api')
+      ),
+    })
+  }
+  const transform = implementation?.transformCss ?? transformCss
+
+  try {
+    const {
+      code,
+      map,
+      exports: moduleExports,
+    } = transform({
+      visitor: createVisitor(options, api, replacements),
+      cssModules: options.modules
+        ? {
+            pattern: process.env.__NEXT_TEST_MODE
+              ? '[name]__[local]'
+              : '[name]__[hash]__[local]',
+          }
+        : undefined,
+      filename: this.resourcePath,
+      code: Buffer.from(source),
+      sourceMap: this.sourceMap,
+      targets: getTargets({ default: userTargets, key: ECacheKey.loader }),
+      inputSourceMap:
+        this.sourceMap && prevMap ? JSON.stringify(prevMap) : undefined,
+      ...opts,
+    })
+    const cssCodeAsString = code.toString()
+
+    if (moduleExports) {
+      for (const name in moduleExports) {
+        if (Object.prototype.hasOwnProperty.call(moduleExports, name)) {
+          exports.push({
+            name,
+            value: moduleExports[name].name,
+          })
+        }
+      }
+    }
+
+    const importCode = getImportCode(imports, options)
+    const moduleCode = getModuleCode(
+      { css: cssCodeAsString, map },
+      api,
+      replacements,
+      options,
+      this
+    )
+    const exportCode = getExportCode(exports, replacements, options)
+
+    const esCode = `${importCode}${moduleCode}${exportCode}`
+    done(null, esCode, map && JSON.parse(map.toString()))
+  } catch (error: unknown) {
+    done(error as Error)
+  }
+}
+
+export const raw = true
+
+const matchRelativePath = /^\.\.?[/\\]/
+
+function isAbsolutePath(str: string) {
+  return path.posix.isAbsolute(str) || path.win32.isAbsolute(str)
+}
+
+function isRelativePath(str: string) {
+  return matchRelativePath.test(str)
+}
+
+// TODO simplify for the next major release
+function stringifyRequest(
```

Comment:
@kdy1 

Next.js has already implemented `stringifyRequest` here: https://github.com/vercel/next.js/blob/9663d78bf9f962a10c901e9ce288aa493c3d3148/packages/next/src/build/webpack/stringify-request.ts#L3-L6

Let's just import that.